src/main/java/com/example/moviemanager/service/tmdbmovieservice/TmdbMovieService.java

```diff
@@ -0,0 +1,78 @@
+package com.example.moviemanager.service.tmdbmovieservice;
+
+import com.example.moviemanager.repository.model.Movie;
+import com.example.moviemanager.service.tmdbmovieservice.configuration.ConfigProperties;
+import com.example.moviemanager.service.tmdbmovieservice.exception.UniqueMovieDetailsNotFoundException;
+import com.example.moviemanager.service.tmdbmovieservice.model.FindMovieInfo;
+import com.example.moviemanager.service.tmdbmovieservice.model.MovieDetails;
+import com.example.moviemanager.service.tmdbmovieservice.model.MovieDetailsResponse;
+import com.example.moviemanager.service.tmdbmovieservice.model.MovieType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.Optional;
+
+@Service
+public class TmdbMovieService {
+
+    private static final String PATH = "/3/search/";
+    private static final String QUERY = "query";
+    private static final String API_KEY = "api_key";
+    private static final String MOVIE_TYPE = "movie";
+    private static final String TV_SHOW_TYPE = "tv";
+    private static final String MOVIE_RELEASE_YEAR = "year";
+    private static final String TV_SHOW_RELEASE_YEAR = "first_air_date_year";
+
+    private final ConfigProperties configProperties;
+    private final RestTemplate restTemplate;
+
+    public TmdbMovieService(ConfigProperties configProperties, RestTemplate restTemplate) {
+        this.configProperties = configProperties;
+        this.restTemplate = restTemplate;
+    }
+
+    public Movie getMovie(FindMovieInfo findMovieInfo) {
+        String endpoint = buildEndpoint(findMovieInfo);
+        MovieDetailsResponse movieDetailsResponse = getResponse(endpoint);
+        MovieDetails movieDetails = retrieveMovieDetails(movieDetailsResponse);
+
+        return buildMovie(findMovieInfo, movieDetails);
```

Comment:
I think this class has too much responsibility. It should be responsible for integrating with TMDB api and returning the response according the query params, but not building some custom response which is logic behind integration. It is your application's specific logic.

Reply:
Refactored. 

TmdbMovieService now returns 'MovieDetails'.